tests/Builder/Stage/SortStageTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Stage;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function MongoDB\object;
+
+/**
+ * Test $sort stage
+ */
+class SortStageTest extends PipelineTestCase
+{
+    public function testAscendingDescendingSort(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::sort(
+                object(
```

Comment:
The `$sort` operator will need some changes to be variadic object [PHPLIB-1269](https://jira.mongodb.org/browse/PHPLIB-1269). But this sort specs are shared with other operators as property. So we also need a SortSpec factory that returns an object.

Reply:
I noticed that in a number of other examples. Ideally, we would be able to write this like so:
```php
Stage::sort(
    age: -1,
    posts: 1,
);
```

IIRC there were some other instances where we pass sort specifications, I'll dig into some pipelines to find those occurrences to see if this applies there as well.